apps/src/lib/levelbuilder/lesson-editor/ResourcesEditor.jsx

```diff
@@ -129,28 +144,43 @@ class ResourcesEditor extends Component {
     this.props.addResource(resource);
   };
 
+  saveEditResource = resource => {
+    this.props.editResource(resource);
+  };
+
   handleRemove = key => {
     this.props.removeResource(key);
   };
 
+  handleEdit = resource => {
+    this.setState({newResourceDialogOpen: true, editingResource: resource});
+  };
+
   handleAddResourceClick = e => {
     e.preventDefault();
     this.setState({newResourceDialogOpen: true});
   };
 
   handleNewResourceDialogClose = () => {
-    this.setState({newResourceDialogOpen: false});
+    this.setState({newResourceDialogOpen: false, editingResource: null});
   };
 
   render() {
     return (
       <div>
-        <AddResourceDialog
-          isOpen={this.state.newResourceDialogOpen}
-          onSave={this.addResource}
-          handleClose={this.handleNewResourceDialogClose}
-          courseVersionId={this.props.courseVersionId}
-        />
+        {this.state.newResourceDialogOpen && (
+          <AddResourceDialog
+            isOpen={this.state.newResourceDialogOpen}
+            onSave={
+              this.state.editingResource
+                ? this.saveEditResource
+                : this.addResource
+            }
+            handleClose={this.handleNewResourceDialogClose}
+            existingResource={this.state.editingResource}
+            courseVersionId={this.props.courseVersionId}
+          />
+        )}
```

Comment:
I am on board with this way of parameterizing the AddResourceDialog. Nice choice!